update platform/bundles explanation and update deploying by hand section (#486)
killbill/killbill-docs: userguide/tutorials/plugin_installation.adoc

* A `sha1.yml` file which is a used by the `KPM` tool to keep track of artifacts that were already downloaded to avoid downloading binaries already present on the filesystem. KPM also offers the `--force-download` to override that behavior.

* A `platform` folder: Also known as *bundles*. This is a directory for deploying Kill Bill's internal OSGI bundles, such as KPM, Logging, Eureka, Metrics, and any JARs/OSGI plugins from `<killbill-platform>/osgi-bundles`. Deploying 3rd party plugins into this directory is not recommended.

> * A `plugins` folder: This is where your plugin or 3rd party plugins deployed. This folder contains:

Reviewer: ```suggestion
* A `plugins` folder: This is where your plugin or 3rd party plugins are deployed. This folder contains:
```

=== Deploying by Hand

> As explained in link:#_overview[overview section], Kill Bill have 2 locations to deploying plugins:

Reviewer: ```suggestion
As explained in link:#_overview[overview section], Kill Bill has two locations to deploying plugins:
```

1. `/platform` (bundles) for Kill Bill internal plugins, and;
2. `/plugins` for 3rd party plugins.

Deploying bundles is straightforward:

> 1. Build `killbill-platform` project.

Reviewer: Does this refer to the https://github.com/killbill/killbill-platform repo? If so, consider adding a link to the repo to make this clear.

Author: fixed

2. Copy desired bundles to `/platform` folder.
3. Start Kill Bill server.

If you want to deploy 3rd party or your own plugins, you need to:

> 1. Create new folder with the name that represents your plugin. It is a good practice to set plugin name with "-plugin" suffix.

Reviewer: ```suggestion
1. Create a new folder with the name that represents your plugin. It is a good practice to set plugin name with "-plugin" suffix.
```

> 2. Add sub-folder that represent version of your plugin. For example, you can set `1.0.0` if it is brand-new plugin.

Reviewer: ```suggestion
2. Add a sub-folder that represents your plugin version.
```

> 3. Put your plugin JAR file into folder you've created in point 2. It is recommended that your plugin have this following format: `<plugin-name>-<semver-plugin-version>.jar`.

Reviewer: ```suggestion
3. Copy your plugin JAR file into the folder you've created in step 2. It is recommended that your plugin jar is named in the following format: `<plugin-name>-<semver-plugin-version>.jar`.
```
